src/tree/mod.rs

```diff
@@ -630,14 +636,15 @@ impl Materializer for BLinkMaterializer {
         vec![Frag::Base(base_node, root)]
     }
 
-    fn recover(&mut self, frag: &Frag) -> Option<PageID> {
+    fn recover(&self, frag: &Frag) -> Option<PageID> {
```

Comment:
recover is now responsible for providing its own interior mutability, because it may be called by any thread that is performing a snapshot, as it "catches up" with the new log tip, from the last snapshot.